Server.java

```diff
@@ -51,6 +56,63 @@ public void run() {
                   }
             });
 
+            if(feature == 'i'){
```

Comment:
like we talked about we should pull all of this out into the DH class which simply returns an established key to the client/server. We don't want the client or server to have to know how DH works under the hood.